guide/guide.md

```diff
@@ -101,6 +101,45 @@ Do not put any spaces around the `.`.
 x.foo().bar().baz(x, y, z);
 ```
 
+### Match
+
+For a single-expression matcher to a single-expression matchee, which fits on
+one line, fit it on one line. If one needs to use an `if`, similarly, if it fits
+on one line, use only one line. If the match pattern does not fit on one line,
+first split before the `=>`, and then split before the `if`. These should all
+use the same indentation, except for `if`, which should be indented once, and
+for the matchee, which should be indented once. The `=>` should not be
+reindented.  When one uses multiple matchers, split by a `|`, put all matchers
+on separate lines, and put the `|` at the beginning of the line.  One may add
+whitespace in order to differentiate between blocks of matchers.
+
+Empty matchees, where one does not want to do anything, shall be denoted by a
+`{}`. Matchees with more than one expression should be surrounded by a `{}`
+block. Place a comma after each match pattern.
+
+
+```rust
+match res {
+  Ok(3) => return IsThree, // simple
+  Ok(4) => if cond {
```

Comment:
There is some [ongoing discussion](https://github.com/rust-lang-nursery/fmt-rfcs/issues/35#issuecomment-273676676) about whether to do this or to do:

```rust
    Ok(4) => {
        if cond {
            ...
        }
    }
```